test/puppetlabs/ssl_utils/core_test.clj

```diff
@@ -253,6 +253,21 @@
                                         "2.5.29.35")]
         (is (= actual-ext expected-ext))))
 
+    (testing (str "signing as an intermediate ca with a type 2 subject id and with a certificate")
+      (let [int-sign-exts (-> (create-ca-extensions (get-public-key issuer-key-pair) (get-public-key key-pair))
+                              (swap-extension (create-truncated-subject-key-identifier(get-public-key key-pair))))
+            int-ca (sign-certificate issuer issuer-priv serial not-before
+                                     not-after subject subj-pub
+                                     int-sign-exts)
+            new-key-pair (generate-key-pair)
+            cert (sign-certificate subject (get-private-key key-pair) serial not-before not-after
+                                   (cn "bar") (get-public-key new-key-pair)
+                                   [(authority-key-identifier int-ca)])
+            int-ca-subject-id (get-extension-value int-ca "2.5.29.14")
+            cert-auth-id (get-extension-value cert "2.5.29.35")]
+        (is (= 8 (count int-ca-subject-id)))
```

Comment:
a comment that type 1 keyids are 20 bytes and type 2 are 8 bytes would be helpful. Maybe putting it behind a constant like `(def type-2-id-byte-length 8)`?